content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+well most of the time, but may fail to guess properly in some cases. The
+`delim` argument is used to specify the delimiter of the file explicitly.
+
+```{r}
+library(vroom)
+
+data <- vroom("flights.tsv")
+```
+
+The summary message after reading also differs from readr. We hope this output
+gives a more informative indication if the types of your columns are being guessed
+properly. However you can still retrieve and print the full column specification
+with `spec()`.
+
+```{r}
+spec(data)
+```
+
+The message will be disabled if you supply a column specification to `col_types` when reading.
+
+```{r}
+s <- spec(data)
+
+data <- vroom("flights.tsv", col_types = s)
+```
+
+## Reading multiple files
+
+One feature new to vroom is built-in support for reading sets of files with the
+same columns into one table. Just pass the filenames to be read directly to
+vroom.
+
+```{r}
+# Split the flights data by carrier
+purrr::iwalk(
+  split(flights, flights$carrier),
+  ~ vroom::vroom_write(.x, glue::glue("flights_{.y}.tsv"), delim = "\t")
+)
+```
+
+Then we can efficiently read all of the files into one tibble by passing the filenames
+directly to `vroom()`.
+
+```{r}
+files <- fs::dir_ls(glob = "flights_*tsv")
+files
+data <- vroom(files)
+```
+
+## Reading and writing compressed files
+
+Just like readr, vroom automatically reads and writes zip, gzip, bz2 and xz compressed
+files with the standard file extensions.
+
+```{r}
+vroom_write(flights, "flights.tsv.gz")
+
+# Check file sizes to show file is compressed
+fs::file_size(c("flights.tsv", "flights.tsv.gz"))
+
+# Read the file back in
+data <- vroom("flights.tsv.gz")
+```
+
+# Reading remote files
+
+vroom can also read files from the internet as well by passing the URL of the file to `vroom()`.
+
+```{r}
+file <- "https://raw.githubusercontent.com/r-lib/vroom/master/inst/extdata/mtcars.csv"
+data <- vroom(file)
+```
+
+It can even read gzipped files from the internet (although currently not the other compressed formats).
+
+## Reading and writing from pipe connections
+
+vroom provides efficient input and output from `pipe()` connections.
+
+This is useful for doing things like pre-filtering large inputs with command line tools like [grep](https://en.wikipedia.org/wiki/Grep).
+
+```{r}
+# Return only flights on United Airlines
+data <- vroom(pipe("grep -w UA flights.tsv"), col_names = names(flights))
+```
+
+Or using multi-threaded compression programs like
+[pigz](https://zlib.net/pigz/), which can greatly reduce the time to write compressed
+files.
+
+```{r}
+bench::workout({
+  vroom_write(flights, "flights.tsv.gz")
+  vroom_write(flights, pipe("pigz > flights.tsv.gz"))
+})
+```
+
+## Column selection
+
+`vroom` introduces a new argument `col_select` which makes selecting columns to
+keep (or omit) more straightforward.
+
+The parameter uses the same interface as `dplyr::select()`, so you can do flexible selection operations.
```

Comment:
I'd specifically state `col_select` again here, rather than "The parameter," just since it's a new paragraph.